dask/array/tests/test_array_core.py

```diff
@@ -2553,11 +2584,13 @@ def test_from_array_scalar(type_):
 def test_from_array_no_asarray(asarray, cls):
     def assert_chunks_are_of_type(x):
         chunks = compute_as_if_collection(Array, x.dask, x.__dask_keys__())
-        for c in concat(chunks):
+        # If it's a tuple of tuples we want to concat, but if it's a tuple
+        # of 1d arrays, we just want to iterate directly
+        for c in concat(chunks) if isinstance(chunks[0], tuple) else chunks:
```

Comment:
Why did this have to change? I'm confused why it worked before but not now.

Reply:
This was the dumbest thing. It was a bad interaction between `concat`, the fact that numpy arrays are iterable, and the difference between a 1xN and and Nx1 numpy array:
```python
from toolz import concat
import numpy

x = numpy.ones((1,2)).T
list(concat((x,x)))   # [array([1.]), array([1.]), array([1.]), array([1.])]

x = numpy.ones((1,2))
list(concat((x,x)))  # [array([1., 1.]), array([1., 1.])]
```
This is really just meant to iterate over chunks, so I was heading off drilling into numpy arrays.

As for why it changed -- some slight variation in denesting in the array finalization that I didn't really track down, and which doesn't actually change the result.